### PR TITLE
AZP/PERF: Pass GitHub PR as param

### PR DESCRIPTION
**Pull request by Alexey-Rivkin**
## What
Resolve the error in perf pipeline:
"fatal: reference is not a tree"

### CODE REVIEW

**Comment by Alexey-Rivkin** [comment]
Working build:
https://dev.azure.com/ucfconsort/ucx/_build/results?buildId=67405&view=logs&j=94be751c-9e4c-5c28-d13e-4c720246b05d&t=af4fd65c-9e75-50b3-0a86-2048b64e0183&l=44
(Depends on pending changes in PerfX)

File: buildlib/azure-pipelines-perf.yml [hunk 1]
```diff
@@ -99,6 +99,7 @@ stages:
               module load /hpc/local/etc/modulefiles/hpcx-gcc
 
               set -xeE -o pipefail
+              export SOURCE_BRANCH=$(Build.SourceBranch)
```
Comment (yosefe) [inline buildlib/azure-pipelines-perf.yml:102]: why needed? who uses this var?

Comment (yosefe) [reply buildlib/azure-pipelines-perf.yml:102]: if it's perfx.py - maube should pass it by cmdline param?

Comment (Alexey-Rivkin) [reply buildlib/azure-pipelines-perf.yml:102]: Used by the PerfX code:
https://github.com/Mellanox-lab/PerfX/pull/53

Comment (Alexey-Rivkin) [reply buildlib/azure-pipelines-perf.yml:102]: Passing as a param is an option, but would require bigger changes at PerfX.
Need to add args parsing, then path it all the way to the library that actually uses it.
I would prefer the current solution.

Comment (yosefe) [reply buildlib/azure-pipelines-perf.yml:102]: but i see other things are passed as parameters: "--config", "--before", "--after"
why SOURCE_BRANCH is different?

Comment (Alexey-Rivkin) [reply buildlib/azure-pipelines-perf.yml:102]: Good point, will do. 
I was expecting Azure built-in vars to be available in ENV, like it works with "AGENT_ID".
That's how I started with using it first and exporting later.

Comment (yosefe) [reply buildlib/azure-pipelines-perf.yml:102]: IMO, PerfX should not expect any Azure env vars, so we could run it also from command line

Comment (Alexey-Rivkin) [reply buildlib/azure-pipelines-perf.yml]: Done

**Review by yosefe — APPROVED** [review]
pls squash

**Review by yosefe — APPROVED** [review]

**Pull request merged**